core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -80,8 +86,59 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
   //periodically emit metrics (don't need to do this for each message!)
   context.system.scheduler.schedule(30.seconds, 10.seconds, self, EmitMetrics)
 
+  // Key is `kind,memory,time`, value is the number of cold Start in minute
+  var coldStartCount = immutable.Map.empty[String, Int]
+
   backfillPrewarms(true)
 
+  // check periodically every 1 minute, delete prewarmed container if unused for sometime
+  context.system.scheduler.schedule(10.seconds, 1.minute, self, DeleteUnusedPrewarmedContainer)
+
+  // check periodically for the cold start and create some increment containers automatically if activation >= threshold
+  context.system.scheduler.schedule(1.minute, 1.minute, self, SupplementPrewarmedContainer)
```

Comment:
Use same scheduled message to do both delete + supplement

Reply:
Already used same message.